error.go

```diff
@@ -88,24 +89,34 @@ func (e *wrappedManyError) Is(target error) bool {
 	return false
 }
 
+func (e *wrappedManyError) As(target interface{}) bool {
+	for _, err := range e.errs {
```

Comment:
Also I don't follow the `As` implementation, I thought we arrived at a different "algorithm" on the call. Basically, for this hierarchy

```
QueryError
|
+---InvalidSyntaxError
```

If `e` is a struct of `invalidSyntaxError`, it should still be possible to cast it into a `QueryError`:

```
var e *QueryError
f errors.As(err, &e) {
 ...
}
```

**without** a `queryError` struct being present in `.errs`.

Reply:
This struct doesn't have anything to do with the EdgeDB error hierarchy. see https://github.com/edgedb/edgedb-go/pull/51#discussion_r553584300

Reply:
That linked comment doesn't fully answer my question -- would the code snippet in my previous comment work with the way the PR reflects errors?

Reply:
> It doesn't require any custom As() implementation.

Ah, I see now, nvm!